### ISSUE TITLE
issue::checkpoints are not saving the information and weights after few days. givinig different results.

### ISSUE THREAD

**Issue by anushka192001** [pytorch/pytorch#125576]
### 📚 The doc issue

when loading a saved pytorch model after few days giving very bad results. 
hamming score suddenly dropped from 75% to 0.1% and flat score dropped from 65% to 0.3%.

torch.save(model, 'models/model_0.pth')
model = torch.load('models/model_0.pth')

checkpoints are not saving the information and weights after few days.

**Comment by rghosh08** [comment]
A few possible fixes:

- Environment Change: Ensure that the environment in which you're loading the model remains consistent. If any dependencies or settings change, it could affect the model's performance.
- Data Drift: If your model is trained on dynamic data, a few days could introduce significant drift in the data distribution. This can cause the model's performance to degrade over time.
- Random Initialization: Some models, especially those using certain types of regularization or random initialization techniques, might exhibit variations in performance due to random seeds. Ensure that your model's initialization is reproducible.
- Model Drift: Just like data drift, the model itself may drift over time. This can happen if the model is too complex for the problem or if the training data doesn't fully represent the test data distribution.
- Memory Corruption: It's possible that the saved model file is getting corrupted over time, leading to unexpected behavior when loaded. Make sure your saving and loading processes are robust and error-free.

Hope this helps.

**Comment by anushka192001** [comment]
@rghosh08  i am using virtual machine with 2 nvidia gpus connected to my visual studio on my local machine. i am training a distillibert on my 22000 dataset and when i saved the trained model and loaded immidiately and tested that time it worked fine but after 2 to 3 days when i loaded in same enivronemnt and dependecies which defenitely didn't changed then i got poor results from that loaded model. 


also when i do traintest split then i use randomseed so it always distributes the data in the same way as train and test. so data is same always.   

model and other ghyperpapramemters also didn't changed they remained same when i saved and loaded the model.

i trianed the model for 2 hours.







these are the important code lines of my file related to this problem

MAX_LEN = 512
TRAIN_BATCH_SIZE = 32
VALID_BATCH_SIZE = 32
EPOCHS = 10
LEARNING_RATE = 1e-05
tokenizer = DistilBertTokenizer.from_pretrained('distilbert-base-uncased', truncation=True, do_lower_case=True)


train_size = 0.8
train_data=new_df.sample(frac=train_size,random_state=200)
test_data=new_df.drop(train_data.index).reset_index(drop=True)
train_data = train_data.reset_index(drop=True)


print("FULL Dataset: {}".format(new_df.shape))
print("TRAIN Dataset: {}".format(train_data.shape))
print("TEST Dataset: {}".format(test_data.shape))

training_set = MultiLabelDataset(train_data, tokenizer, MAX_LEN)
testing_set = MultiLabelDataset(test_data, tokenizer, MAX_LEN)



train_params = {'batch_size': TRAIN_BATCH_SIZE,
                'shuffle': True,
                'num_workers': 0
                }

test_params = {'batch_size': VALID_BATCH_SIZE,
                'shuffle': True,
                'num_workers': 0
                }

training_loader = DataLoader(training_set, **train_params)
testing_loader = DataLoader(testing_set, **test_params) 



import torch.nn.functional as F

# Creating the customized model, by adding a drop out and a dense layer on top of distil bert to get the final output for the model. 

class DistilBERTClass(torch.nn.Module):
    def __init__(self):
        super(DistilBERTClass, self).__init__()
        self.l1 = DistilBertModel.from_pretrained("distilbert-base-uncased")
         
        for name, param in self.l1.named_parameters():
                     param.requires_grad = True
             

        self.pre_classifier = torch.nn.Linear(768, 768)
        self.dropout = torch.nn.Dropout(0.1)
        self.classifier = torch.nn.Linear(768, 26)

    def forward(self, input_ids, attention_mask, token_type_ids):
        output_1 = self.l1(input_ids=input_ids, attention_mask=attention_mask)
        hidden_state = output_1[0]
        pooler = hidden_state[:, 0]
       # pooler = torch.mean(hidden_state, dim=1)
        pooler = self.pre_classifier(pooler)  
        pooler = torch.nn.Tanh()(pooler)
        pooler = self.dropout(pooler)
        output = self.classifier(pooler) 
        output = F.sigmoid(output)
        return output

model = DistilBERTClass()
model.to(device)


def loss_fn(outputs, targets):
    return torch.nn.BCELoss()(outputs, targets)  


optimizer = torch.optim.Adam(params =  model.parameters(), lr=LEARNING_RATE)




def save_checkpoint(model, optimizer, epoch, iteration, loss, filepath='checkpoint.pth'):
    checkpoint = {
        'epoch': epoch,
        'iteration': iteration,
        'model_state_dict': model.state_dict(),
        'optimizer_state_dict': optimizer.state_dict(),
        'loss': loss
    }
    torch.save(checkpoint, filepath)





# Create an instance of your model
model = DistilBERTClass()
model.to(device)

# Define your optimizer (make sure the settings match those used during training)
optimizer = torch.optim.Adam(model.parameters(), lr=LEARNING_RATE)
 
# Load the checkpoint file
checkpoint = torch.load('checkpoint.pth')
    
# Load the model's state_dict from the checkpoint
model.load_state_dict(checkpoint['model_state_dict'])

# Load the optimizer's state_dict from the checkpoint
optimizer.load_state_dict(checkpoint['optimizer_state_dict'])

# Set other parameters from the checkpoint if needed
epoch = checkpoint['epoch']
iteration = checkpoint['iteration']
loss = checkpoint['loss'] 




def validation(testing_loader):
    model.eval()
    fin_targets=[]
    fin_outputs=[]   
    with torch.no_grad():
        for _, data in tqdm(enumerate(testing_loader, 0)):
            ids = data['ids'].to(device, dtype = torch.long)
            mask = data['mask'].to(device, dtype = torch.long)
            token_type_ids = data['token_type_ids'].to(device, dtype = torch.long)
            targets = data['targets'].to(device, dtype = torch.float)
            outputs = model(ids, mask, token_type_ids)
            fin_targets.extend(targets.cpu().detach().numpy().tolist())
            fin_outputs.extend(outputs.cpu().detach().numpy().tolist()) 
    return fin_outputs, fin_targets



outputs, targets = validation(testing_loader)
final_outputs = np.array(outputs) >=0.5
(then the accuracy suddenly dropped from 65% to 0.1% when i loaded the model again in 2-3 days)
val_flat_accuracy_score = flat_accuracy(np.array(targets), np.array(final_outputs))

**Comment by anushka192001** [comment]
@rghosh08 thanks for reaching out can you please check the mail i more thoroughly describled my problem

**Comment by jbschlosser** [comment]
I encourage you to post on the [PyTorch forums](https://discuss.pytorch.org/), as you'll have more luck receiving assistance there. Note that the issues here are reserved for problems within PyTorch itself. Closing for now but please feel free to reopen with details on reproduction if discussion on the forums identifies a PyTorch bug.
